src/utils/discord.ts

```diff
@@ -0,0 +1,82 @@
+import discord from 'discord.js';
+import { TournamentId } from '.prisma/client';
+import { fetchTournaments, fetchTournament } from '../operations/tournament';
+import { fetchTeams } from '../operations/team';
+import env from './env';
+
+// Create bot as a discord client
+const bot = new discord.Client();
+
+async function createChannel(channelName: string, channelType: 'voice' | 'text', tournamentId: TournamentId) {
+  // Get channel and create channels in it
+  const server = bot.guilds.cache.get(env.discord.server);
+  const channel = await server.channels.create(channelName, {
+    type: channelType,
+  });
+
+  const categoryId = (await fetchTournament(tournamentId)).discordCategoryId;
+  await channel.setParent(categoryId);
+
+  // Configure channel's permissions
+  await channel.createOverwrite(channel.guild.roles.everyone, {
+    VIEW_CHANNEL: false,
+  });
+
+  await channel.createOverwrite(
+    channel.guild.roles.cache.find((r) => r.name === channelName),
+    {
+      VIEW_CHANNEL: true,
+    },
+  );
+}
+
+function createRole(teamName: string) {
+  // Get the server and create a role in it
+  const server = bot.guilds.cache.get(env.discord.server);
+  server.roles.create({
+    data: {
+      name: teamName,
+      color: 'FFEBAB',
+    },
+  });
+}
+
+export function setupTeam(team: string, tournamentId: TournamentId) {
+  Promise.all([
```

Comment:
```suggestion
  return Promise.all([
```
Le return est important ici

Reply:
le return est important même si j'ai transervé le `function setupTeams` en `const setupTeams = async(...)` ?